src/agent.rs

```diff
@@ -170,9 +171,9 @@ impl Agent {
         poll.register(&self.child, STATUS, Ready::readable(), PollOpt::level())
             .unwrap();
         let mut events = Events::with_capacity(1);
-        poll.poll(&mut events, None).unwrap();
-
-        let output = self.child.try_recv().unwrap();
+        poll.poll(&mut events, Some(Duration::new(5, 0))).unwrap();
```

Comment:
Can you add a comment somewhere prominent that there's a 5 seconds timeout on the polling. Just in case the number is insufficient or causes any other intermittent issues.

Reply:
Done.